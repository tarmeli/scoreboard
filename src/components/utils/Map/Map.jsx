import React from "react";

const WrapWithKey = ({ children }) => children;

const Map = ({ items, children: toChild }) => {
    return items.map((item, index) => (
        <WrapWithKey key={index}>{toChild(item)}</WrapWithKey>
    ));
};

export default Map;
