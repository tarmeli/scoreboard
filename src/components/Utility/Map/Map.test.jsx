import React from "react";
import { mount } from "enzyme";

import Map from "./Map";

describe("Map", () => {
    const itemsStub = [
        { itemProperty: "some-item-property" },
        { itemProperty: "some-item-property" },
    ];

    it("renders", () => {
        const component = mount(
            <Map items={itemsStub}>
                {({ itemProperty }) => <div>{itemProperty}</div>}
            </Map>
        );

        expect(component).toMatchSnapshot();
    });
});
