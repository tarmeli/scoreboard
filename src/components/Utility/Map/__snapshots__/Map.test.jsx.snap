// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders 1`] = `
<Map
  items={
    Array [
      Object {
        "itemProperty": "some-item-property",
      },
      Object {
        "itemProperty": "some-item-property",
      },
    ]
  }
>
  <WrapWithKey
    key="0"
  >
    <div>
      some-item-property
    </div>
  </WrapWithKey>
  <WrapWithKey
    key="1"
  >
    <div>
      some-item-property
    </div>
  </WrapWithKey>
</Map>
`;
